### PR TITLE
Modifying certain tables

### PR DESCRIPTION
**Pull request by sudheshna-donthineni**

### CODE REVIEW

File: lib/dbservice/enrollment_record/enrollment_record.ex [hunk 1]
```diff
@@ -25,22 +24,21 @@ defmodule Dbservice.EnrollmentRecords.EnrollmentRecord do
...
-    |> validate_date_of_enrollment
+    |> validate_required([:user_id, :group_id, :group_type, :start_date, :academic_year])
+    |> validate_dates_of_enrollment
   end
 
-  defp validate_date_of_enrollment(changeset) do
-    if get_field(changeset, :date_of_enrollment) != nil do
-      invalidate_future_date(changeset, :date_of_enrollment)
+  defp validate_dates_of_enrollment(changeset) do
+    if get_field(changeset, :start_end, :end_date) != nil do
```
Comment (suryabulusu) [inline lib/dbservice/enrollment_record/enrollment_record.ex:40]: start_date

File: lib/dbservice/enrollment_record/enrollment_record.ex [hunk 2]
```diff
@@ -25,22 +24,21 @@ defmodule Dbservice.EnrollmentRecords.EnrollmentRecord do
...
+      validate_start_end_datetime(changeset, :start_end, :end_date)
```
Comment (suryabulusu) [inline lib/dbservice/enrollment_record/enrollment_record.ex:41]: start_date

File: lib/dbservice/sessions/session.ex [hunk 3]
```diff
@@ -26,16 +26,17 @@ defmodule Dbservice.Sessions.Session do
     field(:platform_id, :string)
     field(:type, :string)
     field(:auth_type, :string)
-    field(:activate_signup, :boolean)
+    field(:signup_form, :boolean)
```
Comment (suryabulusu) [inline lib/dbservice/sessions/session.ex:29]: why is activate signup removed

File: lib/dbservice/sessions/session.ex [hunk 4]
```diff
@@ -60,22 +61,24 @@ defmodule Dbservice.Sessions.Session do
...
       :redirection,
       :pop_up_form,
-      :number_of_fields_in_pop_form,
-      :form_schema_id
+      :popup_form_id,
+      :signup_form_id
     ])
     |> validate_required([
       :name,
+      :session_id,
```
Comment (suryabulusu) [inline lib/dbservice/sessions/session.ex:73]: i dont think session_id should be part of. validation.. when we create sessions on session-creator, initially there is no session id. later on, a lambda function creates session id and updates db rows

File: lib/dbservice_web/controllers/group_controller.ex [hunk 5]
```diff
@@ -11,27 +11,38 @@ defmodule DbserviceWeb.GroupController do
...
         required: false,
-        name: "locale"
+        name: "child_id"
       )
     end
 
     response(200, "OK", Schema.ref(:Groups))
   end
 
+  @spec index(Plug.Conn.t(), any) :: Plug.Conn.t()
```
Comment (suryabulusu) [inline lib/dbservice_web/controllers/group_controller.ex:45]: whats this line 😨

**Review by suryabulusu — COMMENTED** [review]
left comments

File: lib/dbservice/batches/batch.ex [hunk 6]
```diff
@@ -14,14 +14,17 @@ defmodule Dbservice.Batches.Batch do
...
-    has_many :group_type, GroupType, foreign_key: :child_id, where: [type: "batch"]
+    belongs_to :program, Program
+
+    has_many :group, Group, foreign_key: :child_id, where: [type: "batch"]
 
     has_many :enrollment_record, EnrollmentRecord,
-      foreign_key: :grouping_id,
-      where: [grouping_type: "batch"]
+      foreign_key: :group_id,
+      where: [group_type: "batch"]
```
Comment (Bahugunajii) [inline lib/dbservice/batches/batch.ex:26]: Why we still have `group_type` here?

File: lib/dbservice/batch_programs.ex
Comment (Bahugunajii) [inline lib/dbservice/batch_programs.ex:1]: Any reason of removing it?

File: lib/dbservice/enrollment_record/enrollment_record.ex [hunk 7]
```diff
@@ -5,18 +5,17 @@ defmodule Dbservice.EnrollmentRecords.EnrollmentRecord do
   import Ecto.Changeset
   import Dbservice.Utils.Util
 
-  alias Dbservice.Users.Student
+  alias Dbservice.Users.User
```
Comment (Bahugunajii) [inline lib/dbservice/enrollment_record/enrollment_record.ex:8]: I think an `enrollment_record` should be of a `student` and not the `user`. What is the reason of associating it with user?

File: lib/dbservice/schools/school.ex [hunk 8]
```diff
@@ -36,8 +36,8 @@ defmodule Dbservice.Schools.School do
       :code,
       :name,
       :udise_code,
-      :type,
-      :category,
+      :gender_type,
+      :af_school_categorycategory,
```
Comment (Bahugunajii) [inline lib/dbservice/schools/school.ex:40]:
```suggestion
      :af_school_category,
```

**Review by Bahugunajii — CHANGES_REQUESTED** [review]

**Review by Bahugunajii — CHANGES_REQUESTED** [review]
How a `product` can be created?

**Review by Bahugunajii — APPROVED** [review]

**Pull request merged**